src/core/lib/executors/Node.ts

```diff
@@ -782,16 +782,20 @@ export default class Node extends Executor<NodeEvents, Config, NodePlugins> {
       delete config.suites;
 
       if (!require.extensions['.ts']) {
-        if (
-          ((config.node &&
-            config.node.suites.some(pattern => pattern.endsWith('.ts'))) ||
-            (config.plugins &&
-              config.plugins.some(plugin => plugin.script.endsWith('.ts')))) &&
-          typeof this.config.node.tsconfig === 'undefined'
-        ) {
-          register();
-        } else if (this.config.node.tsconfig) {
+        if (this.config.node.tsconfig) {
           register({ project: this.config.node.tsconfig });
+        } else if (typeof this.config.node.tsconfig === 'undefined') {
+          // auto-configure typescript support if we detect its presence
+          if (
+            (config.node &&
+              config.node.suites.some(pattern => pattern.endsWith('.ts'))) ||
+            config.functionalSuites.some(pattern => pattern.endsWith('.ts')) ||
+            (config.plugins &&
+              config.plugins.some(plugin => plugin.script.endsWith('.ts'))) ||
+            existsSync(join(config.basePath, 'tsconfig.json'))
```

Comment:
We should handle this case (the existence of `<config.basePath>/tsconfig.json`) separately since `config.basePath` may not be the project root. In that case, we would need to pass a `project` to the register command.